alchemist/alchemist-engine/src/main/java/it/unibo/alchemist/core/implementations/Engine.java

```diff
@@ -81,6 +81,7 @@
     private Time currentTime = DoubleTime.ZERO_TIME;
     private long currentStep;
     private Thread myThread;
+    private Reaction<T> mu;
```

Comment:
Get this file from the mainline

Reply:
I think all of the changes in this file are needed. In particular the function updateMonitors is needed to, as the name implies, correctly update the monitors.

Reply:
I think all of the changes in this file are needed. In particular the function updateMonitors is needed to, as the name implies, correctly update the monitors.

Reply:
All monitors are correctly updated in the develop version. So I do not see why there should be any difference in the engine between develop and the fxui branch. Can you expand why these changes are needed? Why is a track of the last reaction required?

Reply:
No, you are right. Forget it